jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -267,46 +255,39 @@ private void release(RetainedBucket bucket, Pool.Entry<RetainableByteBuffer> ent
 
         RetainableByteBuffer buffer = entry.getPooled();
         BufferUtil.reset(buffer.getByteBuffer());
-        boolean direct = buffer.isDirect();
-        int capacity = bucket.getCapacity();
-        long excessMemory = addMemoryAndGetExcess(bucket, direct);
-        if (excessMemory > 0)
-        {
-            bucket.recordEvict();
-            // If we cannot free enough space for the entry, remove it.
-            if (!evict(excessMemory, bucket, direct))
-            {
-                subtractMemory(capacity, direct);
-                bucket.recordRemove();
-                entry.remove();
-                return;
-            }
-        }
 
         // We have enough space for this entry, pool it.
         if (entry.release())
+        {
+            if (buffer instanceof Buffer b && b.use() % 100 == 1)
```

Comment:
If `b.use()` returns a negative value, the result of `%100` will also be negative. You should test ` == 0` to avoid that.